### PR TITLE
Deleting SSO Focal Use case

### PR DESCRIPTION
**Pull request by philarcher**
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/did-use-cases/pull/144.html" title="Last updated on Feb 12, 2021, 4:37 PM UTC (a705d82)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-use-cases/144/862f455...a705d82.html" title="Last updated on Feb 12, 2021, 4:37 PM UTC (a705d82)">Diff</a>

### CODE REVIEW

**Comment by agropper** [comment]
I see problems with this use-case but are we discussing SSO and DID in some other context?

**Comment by philarcher** [comment]
@jandrieu and I realized that the Single Sign On Focal use Case had not been written - there are just a few notes. However, the new Correlation-controlled Services use case covers a lot of the key features of SSO. We therefore wish to simply remove the (unwritten) SSO Focal Use Case.

**Comment by philarcher** [comment]
Our comments crossed in the post @agropper - you're too fast! Hopefully you saw what amounts to a reply? We suggest the issue *is* covered.

**Comment by agropper** [comment]
It's my fault for ignoring the focal use case. 

I agree that Correlation-controlled Services provides coverage but I fear that we may bury the lede which is SSO.

The activity around DIDauth and SIOP is already very hard to follow and really deserves it's own focal use-case. 

I originally posted the SSO focal use-case as a direct reflection of our implementation experience with early SSI wallets. Self-sovereign patients need to allow self-sovereign doctors to sign-in to their self-sovereign health record without asking the doctor to set a password for each of their patients. The doctor uses a VC in their wallet as the authentication (SSO) credential. The patient's self-sovereign agent, as relying party, keeps an "allow list" of credential issuers.

I suggest we confer with one or two of the DIDauth and SIOP editors to ensure we have captured their intent in a paragraph like the one above. If you agree, and maybe you can suggest a collaborator, I will try to fix the SSO focal use case pronto.

**Comment by philarcher** [comment]
Adrian, rather than a focal use case, the quickest option at this stage would be simply to massage your brief paragraph a little into a short use case. That's something we can add easily enough. Modulo any comment or further amendments from @jandrieu, in my mind this is the *only* thing between the current draft and declaring it done and handing over to the chairs - which we're keen to do within the coming days.

How's this as a starter:

Wang Qian is receiving treatment from her doctor, Liu Ning, who needs to update his patient's health record. By exchanging DIDs through simply bringing their mobile devices into close enough proximity for the data to be transmitted using <abbr title="Near Field Communication">NFC</abbr>, they're able to establish a private connection between doctor and patient. Miss Wang is able to request and verify Dr Liu's medical credential from his electronic wallet, and thus authorize him to update her medical record. This is all achieved securely without either doctor or patient needing to sign into any system with a password.

**Comment by agropper** [comment]
@philarcher I agree with your suggestion but don't accept your framing of the SSO as an in-person event.

In-person interactions are a tiny, maybe insignificant, fraction of the SSO requirement. Also, this in-person use-case is somewhat covered by ambient authentication now named: Correlation-controlled Services https://w3c.github.io/did-use-cases/#correlation 

SSO, including OpenID Connect (may it rest in peace) is primarily an online experience. The doctor presents credentials and the patient is nowhere to be seen and is not represented by her user agent. Wang Quian does not need a self-sovereign system for SSI to work. She's the relying party in this case and she has almost certainly delegated access control to an institutional system. The doctor, for his part, is not likely to want to wait for the patient to be contacted proper to access. 

How's this:

Wang Qian is receiving a vaccine from her pharmacist, Liu Ning, who needs to update his patient's health record at some hospital. Liu Ning has entered the patient's insurance ID into the pharmacy system and that system is displaying a QR code derived from the hospital that holds Wang Quien's health record. By scanning the QR code, Liu Ning is able to present his pharmacy license credential to the custodian of patient's record. Miss Wang's health record custodian is able to request and verify Dr Liu's medical credential from his electronic wallet, and thus authorize him to update her medical record. This is all achieved securely without either doctor or patient needing to sign into any system with a password.

**Review by jandrieu — APPROVED** [review]

**Comment by agropper** [comment]
LGTM

**Comment by jandrieu** [comment]
@agropper and I had a long conversation this weekend about how we might turn his suggestion into a use case. We talked about whether this is a federation authorization use case rather than SSO and the importance of "break-the-glass" scenarios. One of the challenging disconnects was how one could (or should) handle a lack of prior consent. 

In the break-the-glass situation, we are enabling extraordinary means to deal with an extraordinary situation. Doing so by a blanket back-door authorization to all members of a federation, e.g., a Doctor who is accredited, doesn't seem to be a use case fit for Decentralized Identifiers--yes, you can do it, but doing so in a way that doesn't create confused deputies is a challenge. 

To my sense, done poorly, that isn't much better than putting a back door in a cryptographic suite, "just in case" we need to get access to the information. Such an approach is controversial and largely considered to help the bad guys just as or more than the good guys. The slippery slope risk is that such "emergency" capabilities become adopted for standard, non-emergency use, which then neutralizes any hoped for security and privacy protections from restricting "breaking the glass" to emergencies.

Adrian and I agree that, due to a desire to close this work up and prepare for publishing, that we'll defer this particular use case to a future conversation.
